crates/kitsune_p2p/transport_quic/src/listener.rs

```diff
@@ -35,6 +123,61 @@ impl ListenerInnerHandler for TransportListenerQuic {
             .map_err(TransportError::other)?;
         Ok(async move { Ok(out) }.boxed().into())
     }
+
+    fn handle_take_connecting(
```

Comment:
Some comments inside this function might be helpful. It's pretty hard to follow what is happing

Reply:
:+1: done